src/main/java/de/ellpeck/actuallyadditions/mod/gen/OreGen.java

```diff
@@ -59,19 +60,32 @@ public void generate(Random random, int chunkX, int chunkZ, World world, IChunkG
         int dimension = world.provider.getDimension();
         if(dimension != -1 && dimension != 1){
             if(world.getWorldType() != WorldType.FLAT && !ArrayUtils.contains(ConfigIntListValues.ORE_GEN_DIMENSION_BLACKLIST.getValue(), world.provider.getDimension())){
-                this.generateDefault(world, random, chunkX*16, chunkZ*16);
+                this.generateDefault(world, random, chunkX, chunkZ);
             }
         }
     }
 
     private void generateDefault(World world, Random random, int x, int z){
         if(ConfigBoolValues.GENERATE_QUARTZ.isEnabled()){
-            this.addOreSpawn(InitBlocks.blockMisc, TheMiscBlocks.ORE_QUARTZ.ordinal(), Blocks.STONE, world, random, x, z, MathHelper.getRandomIntegerInRange(random, 5, 8), 10, QUARTZ_MIN, QUARTZ_MAX);
+            this.addOreSpawn(InitBlocks.blockMisc, TheMiscBlocks.ORE_QUARTZ.ordinal(), Blocks.STONE, world, random, x*16, z*16, MathHelper.getRandomIntegerInRange(random, 5, 8), 10, QUARTZ_MIN, QUARTZ_MAX);
         }
 
-        if(ConfigBoolValues.GEN_LUSH_CAVES.isEnabled() && random.nextInt(ConfigIntValues.LUSH_CAVE_CHANCE.getValue()) <= 0){
-            BlockPos posAtHeight = world.getTopSolidOrLiquidBlock(new BlockPos(x+random.nextInt(16)+8, 0, z+random.nextInt(16)+8));
-            this.caveGen.generate(world, random, posAtHeight.down(MathHelper.getRandomIntegerInRange(random, 15, posAtHeight.getY()-15)));
+        if(ConfigBoolValues.GEN_LUSH_CAVES.isEnabled()){
+            StructureBoundingBox box = new StructureBoundingBox(x*16+8, 0, z*16+8, x*16+8+15, 255, z*16+8+15);
+            int chunkRadius = 1;
+            for(int dx = -chunkRadius; dx <= chunkRadius; dx++) {
+                for(int dz = -chunkRadius; dz <= chunkRadius; dz++) {
+                    int chunkX = x+dx;
+                    int chunkZ = z+dz;
+                    int randConst = 0x969ce69d;//so that it won't generate the same numbers as other mod that does the same thing
+                    Random chunkRand = new Random(randConst ^ world.getSeed() ^ (chunkX*29+chunkZ*31));
```

Comment:
If I didn't use chunk positions: for a given world, either all chunks would have lush caves or none of them. The seed wouldn't depend on chunk positions
If I didn't use world seed: each world would have the same lush cave positions, because it wouldn't depend on world seed
If I didn't add that weird random number: if other mod used the same way to generate random numbers, it would generate the same positions